### PR TITLE
[GLUTEN-3213][CH] use wrong type(int32) to construct null bitmap mask when r2c

### PR DESCRIPTION
**Pull request by lhuang09287750**
## What changes were proposed in this pull request?

change it to long(int64)

(Fixes: \#3213)

## How was this patch tested?

add a new testcase in GlutenDataFrameAggregateSuite

### CODE REVIEW

File: cpp-ch/local-engine/Parser/CHColumnToSparkRow.cpp [hunk 1]
```diff
@@ -77,7 +77,7 @@ void bitSet(char * bitmap, size_t index)
 ALWAYS_INLINE bool isBitSet(const char * bitmap, size_t index)
 {
     assert(index >= 0);
-    int64_t mask = 1 << (index & 63);
+    int64_t mask = 1L << (index & 63);
```
Comment (zhanglistar) [inline cpp-ch/local-engine/Parser/CHColumnToSparkRow.cpp:80]: Use 1LL will be better. Since in x86_32, long is 32bit.

**Review by taiyang-li — APPROVED** [review]
LGTM.

File: cpp-ch/local-engine/Parser/CHColumnToSparkRow.cpp (hunk 1 above)
Comment (lhuang09287750) [reply cpp-ch/local-engine/Parser/CHColumnToSparkRow.cpp:80]: err, I had no idea about the difference between L and LL. 
In the function bitSet(just before isBitSet) ,1L is used to make mask
And in the BItSetMethods.class which is in spark-unsafe jar ,1L is also used in isSet function,like this:
```
  public static boolean isSet(Object baseObject, long baseOffset, int index) {
    assert index >= 0 : "index (" + index + ") should >= 0";

    long mask = 1L << (index & 63);
    long wordOffset = baseOffset + (long)(index >> 6) * 8L;
    long word = Platform.getLong(baseObject, wordOffset);
    return (word & mask) != 0L;
  }
```
So I just use L for consistency. Seems LL is better

**Review by baibaichen — APPROVED** [review]
LGTM

**Pull request merged**